lib/core/src/lib/common/services/user-preferences.service.ts

```diff
@@ -246,11 +247,15 @@ export class UserPreferencesService {
 
     private getLanguageByKey(key: string): LanguageItem {
         const defaultLanguage = { key: 'en' } as LanguageItem;
+        let language: LanguageItem;
 
-        const registeredLanguages = this.appConfig.get<Array<LanguageItem>>(AppConfigValues.APP_CONFIG_LANGUAGES_KEY);
-        if (registeredLanguages && Array.isArray(registeredLanguages)) {
-            return registeredLanguages.find((language) => key.includes(language.key)) || defaultLanguage;
+        const customLanguages = this.appConfig.get<Array<LanguageItem>>(AppConfigValues.APP_CONFIG_LANGUAGES_KEY);
+        if (customLanguages && Array.isArray(customLanguages)) {
```

Comment:
no need for both conditions, Array.isArray is enough because if customLanguages will be null or undefined then isArray gives just false because null and undefined aren't arrays.

Reply:
The condition was present before as well. However, since the suggestion does add value, and removes unnecessary code, updated the condition